src/Umbraco.Core/Manifest/ManifestContentAppDefinition.cs

```diff
@@ -112,6 +115,18 @@ public ContentApp GetContentAppFor(object o)
                 // else iterate over each entry
                 foreach (var rule in rules)
                 {
+                    if (rule.PartA == "role")
+                    {
+                        foreach (var group in userGroups)
+                        {
+                            if (rule.Matches(rule.PartA, group.Alias))
```

Comment:
need a bit of time to think about that one - not sure that's the way it works - if you're hitting a rules that matches and says, "don't show" (eg `-role/editor`) then you simply cannot show the content app (ie you need to return null) - that's how it works with contentType aliases. if no role rule is provided, then all roles are ok, but if one single `+role/editor` rule is provided then other roles are excluded. thats also how it works with contentType aliases. so I think the logic is a little bit more complex here.

ah and then... we should unit-test these rules I guess ;-)

Reply:
Thanks @zpqrtbnk :) You are right, I need to think of a few more scenarios. I will have a look and hopefully resubmit PR.